### PR TITLE
Add pre-trained models for ResNet

### PR DESCRIPTION
**Pull request by adamjstewart**
Still need to add tests, but this is basically what I'm thinking for pre-trained models. It's almost a carbon-copy of how torchvision does things, but with an added "sensor" field that lets you specify which satellite you want your model trained on.

Some things to consider:

* Do we want to support model + sensor + task combinations? I'm thinking not since torchvision only supports model and not model + task
* Do we want to support arbitrary band subsets? We could support Sentinel 2 "all bands" and "rgb" but I don't think we want to support any more than we have to
* Do we want error checking or some way to let users know which models have been trained for which sensors?

### CODE REVIEW

File: torchgeo/models/resnet.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def _resnet(sensor: str, arch: str, block: Type[Union[BasicBlock, BottleNeck]], layers: List[int], pretrained: bool, progress: bool, **kwargs: Any) -> ResNet:
+    """ResNet model.
+
+    If you use this model in your research, please cite the following paper:
+
+    * https://arxiv.org/pdf/1512.03385.pdf
+
+    Args:
```
[the remark this answers is not in the document]
Comment (adamjstewart) [reply torchgeo/models/resnet.py:35]: So far we haven't seen any evidence that RGB-only satellite imagery pretraining works any better than ImageNet pretraining. Until then, I think we should just use ImageNet pretrained weights for RGB-only imagery.

[the remark this answers is not in the document]
Comment (calebrob6) [reply torchgeo/models/resnet.py:39]: Not sure where my emailed comment went... But, we do indeed have that evidence from the SeCo paper https://arxiv.org/pdf/2103.16607.pdf, and will definitely want the distinction.

image

[the remark this answers is not in the document]
Comment (adamjstewart) [reply torchgeo/models/resnet.py:39]: But do we plan to train models for Sentinel-2 all bands and RGB-only? What other band combinations will we train? The more we add, the more compute time we will require to update all models.

[the remark this answers is not in the document]
Comment (calebrob6) [reply torchgeo/models/resnet.py:39]: Yes, definitely those two combinations, maybe more -- I can imagine RGB + one of the IR bands

[the remark this answers is not in the document]
Comment (adamjstewart) [reply torchgeo/models/resnet.py:42]: Done

**Review by calebrob6 — APPROVED** [review]

**Pull request merged**